### ISSUE TITLE
Provide more control for tree searching

### ISSUE THREAD

**Issue by willeastcott** [playcanvas/pcui#124]
Currently, searching tree items does a fuzzy match.

https://github.com/playcanvas/pcui/blob/master/src/helpers/search.js#L3

Sometimes, it is useful to just search back on exact matches.

Allowing the developer/end-user the ability to select a match type would be useful.

**Comment by Maksims** [comment]
Indeed. I believe better sorting of search results - would do a great job here:
1. Exact matches first (case insensitive)
2. Partial matches then (case insensitive)
3. Fuzzy matches based on string distance then
